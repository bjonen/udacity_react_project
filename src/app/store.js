// We just name the default export "middleware" and import it as middleware.
import middleware from "../middleware";
import { configureStore } from "@reduxjs/toolkit";
import logger from "../middleware/logger.js";
import { apiSlice } from "../apiSlice.js";

const reducer = (state = {}, action) => {
  return state;
};

// https://redux.js.org/tutorials/quick-start#create-a-redux-store
// We use configureStore instead of createStore which is deprecated.
// const store = createStore(reducer, middleware);
// https://redux-toolkit.js.org/api/configureStore
export default configureStore({
  reducer: {
    [apiSlice.reducerPath]: apiSlice.reducer,
  },
  // Will be passed to applyMiddleware
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware().concat(apiSlice.middleware),
  //   preloadedState: {
  //     counter: 1,
  //   },
});
