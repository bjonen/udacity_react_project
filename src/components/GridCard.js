// Data is passed via OverviewPage. No need for accessing external state.

import Card from "@mui/material/Card";
import CardContent from "@mui/material/CardContent";
import Typography from "@mui/material/Typography";
import CardHeader from "@mui/material/CardHeader";
import Avatar from "@mui/material/Avatar";
import NorthEastSharpIcon from "@mui/icons-material/NorthEastSharp";
import IconButton from "@mui/material/IconButton";
import { useNavigate } from "react-router-dom";
import Box from "@mui/material/Box";

const GridCard = ({ card }) => {
  // XXX Could also pass only question id and let GridCard fetch the
  // the required card information from the store.
  const navigate = useNavigate();

  return (
    <Card elevation={1}>
      <CardHeader
        action={
          <IconButton onClick={() => navigate(`/questions/${card.id}`)}>
            <NorthEastSharpIcon />
          </IconButton>
        }
        title={card.author}
        subheader={card.timestamp}
        avatar={<Avatar src={card.avatar} />}
      />
      <CardContent variant="body2" color="textSecondary">
        <Typography>
          <Box sx={{ fontSize: 10, m: 1 }}>
            1){" "}
            {card.textOptionOne.length <= 40
              ? card.textOptionOne.slice(0, 40)
              : card.textOptionOne.slice(0, 40) + "..."}
          </Box>
          <Box sx={{ fontSize: 10, m: 1 }}>
            2){" "}
            {card.textOptionTwo.length <= 40
              ? card.textOptionTwo.slice(0, 40)
              : card.textOptionTwo.slice(0, 40) + "..."}
          </Box>
        </Typography>
      </CardContent>
    </Card>
  );
};

export default GridCard;
