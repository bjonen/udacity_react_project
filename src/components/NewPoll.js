import { useState } from "react";

const NewPoll = () => {
  const [text1, setText1] = useState("");
  const [text2, setText2] = useState("");

  const handleSubmit = (e) => {
    e.preventDefault();
    setText1("");
    setText2("");
    // Perform submission logic
  };

  const handleChange = (e) => {
    const text = e.target.value;
    const name = e.target.name;
    if (name === "text1") {
      setText1(text);
    } else {
      setText2(text);
    }
  };

  return (
    <div
      style={{
        // https://stackoverflow.com/a/33049392
        display: "flex",
        justifyContent: "center",
        flexDirection: "column",
        //alignSelf: "center",
        alignItems: "center",
      }}
    >
      <div>
        <h1>Create New Poll</h1>
        <span>Would You Rather</span>
        <br />
        <div>
          <form onSubmit={handleSubmit}>
            <label>
              Option A
              <input
                style={{ marginLeft: "10px" }}
                name="text1"
                type="text"
                value={text1}
                onChange={handleChange}
              />
            </label>
            <br />
            <label>
              Option B
              <input
                style={{ marginLeft: "10px" }}
                name="text2"
                type="text"
                value={text2}
                onChange={handleChange}
              />
            </label>
            <br />
            <button className="abc" type="submit">
              Submit
            </button>
          </form>
        </div>
      </div>
    </div>
  );
};

export default NewPoll;
