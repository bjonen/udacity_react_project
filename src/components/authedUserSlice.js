import { createSlice } from "@reduxjs/toolkit";

export const authedUserSlice = createSlice({
  name: "authedUser",
  initialState: {
    id: "sarahedo",
  },
  reducers: {
    login(state, action) {
      state.id = action.payload;
    },
    logout(state, action) {
      state.id = null;
    },
  },
});

export const { login, logout } = authedUserSlice.actions;

// authedUserSlice.reducer;
