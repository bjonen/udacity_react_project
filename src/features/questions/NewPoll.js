// State:
// internal: text1, text2
// external:
// * authedUser (To save the question to the user's list of questions),
// * questions (to check whether the question already exists and to save the new questions)

import { useState } from "react";
import Button from "@mui/material/Button";
import { useSaveQuestionMutation } from "./questionSlice";
import { useSelector } from "react-redux";
import { useNavigate } from "react-router-dom";

const NewPoll = () => {
  const [text1, setText1] = useState("");
  const [text2, setText2] = useState("");
  const [saveQuestion, { isLoading }] = useSaveQuestionMutation();
  const authedUser = useSelector((state) => state.authedUser.id);
  const navigate = useNavigate();
  if (authedUser === "anonymous") {
    return (
      <div style={{ padding: "10px" }}>Please log in to create a new poll</div>
    );
  }

  const handleSubmit = (e) => {
    e.preventDefault();
    setText1("");
    setText2("");
    // Perform submission logic
    saveQuestion({
      optionOneText: text1,
      optionTwoText: text2,
      author: authedUser,
    });
    navigate("/");
  };

  const handleChange = (e) => {
    const text = e.target.value;
    const name = e.target.name;
    if (name === "text1") {
      setText1(text);
    } else {
      setText2(text);
    }
  };

  return (
    <div>
      {isLoading ? (
        <div data-testid="api-loading">Waiting for API request to complete</div>
      ) : (
        <div
          style={{
            // https://stackoverflow.com/a/33049392
            display: "flex",
            justifyContent: "center",
            flexDirection: "column",
            //alignSelf: "center",
            alignItems: "center",
          }}
        >
          <div>
            <h1>Create New Poll</h1>
            <span>Would You Rather</span>
            <br />
            <div>
              <form onSubmit={handleSubmit}>
                <label>
                  Option A
                  <input
                    style={{ marginLeft: "10px" }}
                    data-testid="text1"
                    name="text1"
                    type="text"
                    value={text1}
                    onChange={handleChange}
                  />
                </label>
                <br />
                <label>
                  Option B
                  <input
                    style={{ marginLeft: "10px" }}
                    data-testid="text2"
                    name="text2"
                    type="text"
                    value={text2}
                    onChange={handleChange}
                  />
                </label>
                <br />
                <Button
                  id="submitNewPoll"
                  data-testid="submit-button"
                  type="submit"
                  variant="outlined"
                  style={{ height: "10%", width: "10%" }}
                >
                  Submit
                </Button>
              </form>
            </div>
          </div>
        </div>
      )}
    </div>
  );
};

export default NewPoll;
