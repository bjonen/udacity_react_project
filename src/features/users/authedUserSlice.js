import { createSlice } from "@reduxjs/toolkit";

export const authedUserSlice = createSlice({
  name: "authedUser",
  initialState: {
    id: "anonymous",
  },
  reducers: {
    login(state, action) {
      state.id = action.payload;
    },
    // In case we want to add more sophisticated auth later
    logout(state, action) {
      state.id = null;
    },
  },
});

export const { login, logout } = authedUserSlice.actions;
