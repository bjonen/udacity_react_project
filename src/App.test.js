import GridCard from "./features/users/GridCard";
import MenuAppBar from "./app/Navbar";
import { render, fireEvent } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";
import store from "./app/store.js";
import NewPoll from "./features/questions/NewPoll.js";
import preview from "jest-preview";
import { login } from "./features/users/authedUserSlice.js";
import { act } from "react-dom/test-utils";

describe("NewPoll", () => {
  it("will render correctly", async () => {
    store.dispatch(login("sarahedo"));
    var component = render(
      <BrowserRouter>
        <Provider store={store}>
          <NewPoll />
        </Provider>
      </BrowserRouter>
    );
    expect(component).toMatchSnapshot();
  });

  it("will submit correctly", async () => {
    store.dispatch(login("sarahedo"));
    var component = render(
      <BrowserRouter>
        <Provider store={store}>
          <NewPoll />
        </Provider>
      </BrowserRouter>
    );
    act(() => {
      var input = component.getByTestId("text1");
      fireEvent.change(input, { target: { value: "Buy a Macbook" } });
      var input = component.getByTestId("text2");
      fireEvent.change(input, { target: { value: "Buy a Windows PC" } });
      var submitButton = component.getByTestId("submit-button");
      submitButton.click();
    });
    //await new Promise((r) => setTimeout(r, 100));
    // While we navigate away from the page, the component itself will still show
    // the loading div.
    expect(component.getByTestId("api-loading")).toBeInTheDocument();
    preview.debug();
  });
});

describe("NavBar", () => {
  it("will render correctly", async () => {
    var component = render(
      <Provider store={store}>
        <MenuAppBar />
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });
});

describe("GridCard", () => {
  it("will render correctly", async () => {
    let card = {
      id: "myid",
      author: "bj",
      avatar: "kdkd",
      textOptionOne: "The first option",
      textOptionTwo: "The second option",
      timestamp: "20220231",
    };
    var component = render(
      <BrowserRouter>
        <Provider store={store}>
          <GridCard key="myid" card={card} />
        </Provider>
      </BrowserRouter>
    );
    expect(component).toMatchSnapshot();
    preview.debug();
  });
});
