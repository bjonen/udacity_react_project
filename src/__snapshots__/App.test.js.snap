// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridCard will render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
      >
        <div
          class="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
        >
          <div
            class="MuiCardHeader-avatar css-1ssile9-MuiCardHeader-avatar"
          >
            <div
              class="MuiAvatar-root MuiAvatar-circular css-1wlk0hk-MuiAvatar-root"
            >
              <img
                class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                src="kdkd"
              />
            </div>
          </div>
          <div
            class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
          >
            <span
              class="MuiTypography-root MuiTypography-body2 MuiCardHeader-title css-et1ao3-MuiTypography-root"
            >
              bj
            </span>
            <span
              class="MuiTypography-root MuiTypography-body2 MuiCardHeader-subheader css-83ijpv-MuiTypography-root"
            >
              20220231
            </span>
          </div>
          <div
            class="MuiCardHeader-action css-sgoict-MuiCardHeader-action"
          >
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="NorthEastSharpIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M9 5v2h6.59L4 18.59 5.41 20 17 8.41V15h2V5H9z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
          color="textSecondary"
          variant="body2"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            <div
              class="MuiBox-root css-no710t"
            >
              1)
               
              The first option
            </div>
            <div
              class="MuiBox-root css-no710t"
            >
              2)
               
              The second option
            </div>
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
      >
        <div
          class="MuiCardHeader-avatar css-1ssile9-MuiCardHeader-avatar"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circular css-1wlk0hk-MuiAvatar-root"
          >
            <img
              class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
              src="kdkd"
            />
          </div>
        </div>
        <div
          class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
        >
          <span
            class="MuiTypography-root MuiTypography-body2 MuiCardHeader-title css-et1ao3-MuiTypography-root"
          >
            bj
          </span>
          <span
            class="MuiTypography-root MuiTypography-body2 MuiCardHeader-subheader css-83ijpv-MuiTypography-root"
          >
            20220231
          </span>
        </div>
        <div
          class="MuiCardHeader-action css-sgoict-MuiCardHeader-action"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="NorthEastSharpIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M9 5v2h6.59L4 18.59 5.41 20 17 8.41V15h2V5H9z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <div
        class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        color="textSecondary"
        variant="body2"
      >
        <span
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          <div
            class="MuiBox-root css-no710t"
          >
            1)
             
            The first option
          </div>
          <div
            class="MuiBox-root css-no710t"
          >
            2)
             
            The second option
          </div>
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`NavBar will render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-k008qs"
      >
        <span
          class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
          role="progressbar"
          style="width: 40px; height: 40px;"
        >
          <svg
            class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
            viewBox="22 22 44 44"
          >
            <circle
              class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
              cx="44"
              cy="44"
              fill="none"
              r="20.2"
              stroke-width="3.6"
            />
          </svg>
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-k008qs"
    >
      <span
        class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
        role="progressbar"
        style="width: 40px; height: 40px;"
      >
        <svg
          class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`NewPoll will render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          style="display: flex; justify-content: center; flex-direction: column; align-items: center;"
        >
          <div>
            <h1>
              Create New Poll
            </h1>
            <span>
              Would You Rather
            </span>
            <br />
            <div>
              <form>
                <label>
                  Option A
                  <input
                    data-testid="text1"
                    name="text1"
                    style="margin-left: 10px;"
                    type="text"
                    value=""
                  />
                </label>
                <br />
                <label>
                  Option B
                  <input
                    data-testid="text2"
                    name="text2"
                    style="margin-left: 10px;"
                    type="text"
                    value=""
                  />
                </label>
                <br />
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
                  data-testid="submit-button"
                  id="submitNewPoll"
                  style="height: 10%; width: 10%;"
                  tabindex="0"
                  type="submit"
                >
                  Submit
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        style="display: flex; justify-content: center; flex-direction: column; align-items: center;"
      >
        <div>
          <h1>
            Create New Poll
          </h1>
          <span>
            Would You Rather
          </span>
          <br />
          <div>
            <form>
              <label>
                Option A
                <input
                  data-testid="text1"
                  name="text1"
                  style="margin-left: 10px;"
                  type="text"
                  value=""
                />
              </label>
              <br />
              <label>
                Option B
                <input
                  data-testid="text2"
                  name="text2"
                  style="margin-left: 10px;"
                  type="text"
                  value=""
                />
              </label>
              <br />
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
                data-testid="submit-button"
                id="submitNewPoll"
                style="height: 10%; width: 10%;"
                tabindex="0"
                type="submit"
              >
                Submit
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
